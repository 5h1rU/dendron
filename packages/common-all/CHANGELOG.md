# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.3.11](https://github.com/dendronhq/dendron/compare/v0.3.10...v0.3.11) (2020-07-13)

**Note:** Version bump only for package @dendronhq/common-all

## [0.3.10](https://github.com/dendronhq/dendron/compare/v0.1.6...v0.3.10) (2020-07-13)

### Features

- auto add nodes when deleted or created outside of dendron ([8c311bd](https://github.com/dendronhq/dendron/commit/8c311bda948a1d54088c49fd70eb65d24af5d68f))
- better schema suggestions ([03656bc](https://github.com/dendronhq/dendron/commit/03656bc007810457cb6846f0d6adacab4a7fbd3a))
- match namespace schemas ([7a67b8b](https://github.com/dendronhq/dendron/commit/7a67b8b2fb7caa1b97ee6d492d2801782abecdf6))
- show node descriptions ([aca86f2](https://github.com/dendronhq/dendron/commit/aca86f2a5fd6ee481f93553693a098db0e322890))
- surface unknown schemas ([d014965](https://github.com/dendronhq/dendron/commit/d0149652c985c69a4b2607984d578902820077f1))
- updated icons for schemas ([21804eb](https://github.com/dendronhq/dendron/commit/21804eba61c8dd49e499edd5d548d9d601224e8e))

## [0.3.9](https://github.com/dendronhq/dendron/compare/v0.3.8...v0.3.9) (2020-07-09)

### Features

- updated icons for schemas ([b7a2d8a](https://github.com/dendronhq/dendron/commit/b7a2d8aa517cf88d7a93d07cd2ef19305e48d069))

## [0.3.7](https://github.com/dendronhq/dendron/compare/v0.3.6...v0.3.7) (2020-07-08)

### Features

- better schema suggestions ([ad74bc0](https://github.com/dendronhq/dendron/commit/ad74bc009e1544319a49689394ab8d6b684f6578))
- show node descriptions ([e08fce9](https://github.com/dendronhq/dendron/commit/e08fce994153e28fe504b85e6d9bc1f5fdd93e20))

## [0.3.6](https://github.com/dendronhq/dendron/compare/v0.3.5...v0.3.6) (2020-07-07)

### Features

- auto add nodes when deleted or created outside of dendron ([a7e1ac9](https://github.com/dendronhq/dendron/commit/a7e1ac9b8a4f7f0592ab1b9f86a7a40182693a73))

# [0.3.0](https://github.com/dendronhq/dendron/compare/v0.2.20...v0.3.0) (2020-07-05)

### Features

- match namespace schemas ([1a960bf](https://github.com/dendronhq/dendron/commit/1a960bf26f8984e541b3eb118f60bdc09d8250fe))
- surface unknown schemas ([9bf4d0e](https://github.com/dendronhq/dendron/commit/9bf4d0e61cce2f76bddae1f686f29474201466cb))
