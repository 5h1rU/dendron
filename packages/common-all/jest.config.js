const jestConfig = require("../../jest.config");

console.log(jestConfig);
module.exports = {
  ...jestConfig
};
