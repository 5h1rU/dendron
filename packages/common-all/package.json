{
  "name": "@dendronhq/common-all",
  "version": "0.1.6",
  "description": "common-all",
  "license": "MIT",
  "repository": "https://github.com/dendronhq/dendron",
  "author": {
    "name": "Kevin Lin",
    "email": "kevin@thence.io",
    "url": "https://kevinslin.com",
    "organization": true
  },
  "keywords": [
    "common"
  ],
  "files": [
    "lib"
  ],
  "main": "lib/index",
  "types": "lib/index",
  "scripts": {
    "clean": "rimraf lib && rimraf coverage",
    "format": "",
    "lint": "",
    "prepublishOnly": "npm run build",
    "prebuild": "npm run clean && npm run format && npm run lint && echo Using TypeScript && tsc --version",
    "build": "tsc --pretty",
    "test": "jest",
    "coverage": "jest --coverage",
    "watch": "npm run build -- --watch",
    "test:watch": "jest --watch"
  },
  "dependencies": {
    "dropbox": "^4.0.30",
    "isomorphic-fetch": "^2.2.1",
    "lodash": "^4.17.15",
    "pino": "^6.3.1",
    "uuid": "^8.1.0",
    "yamljs": "^0.3.0"
  },
  "devDependencies": {
    "@types/jest": "^23.3.3",
    "@types/lodash": "^4.14.152",
    "@types/node": "^10.11.4",
    "@types/pino": "^6.0.1",
    "@types/uuid": "^8.0.0",
    "@types/yamljs": "^0.2.31",
    "coveralls": "^3.0.2",
    "jest": "^23.6.0",
    "prettier": "^1.14.3",
    "rimraf": "^2.6.2",
    "ts-jest": "^23.10.3",
    "ts-node": "^7.0.1",
    "tslint": "^5.11.0",
    "tslint-config-prettier": "^1.15.0",
    "typescript": "^3.1.1"
  },
  "engines": {
    "node": ">=10.0.0"
  },
  "jest": {
    "preset": "ts-jest"
  },
  "gitHead": "403eb012f8b34bc115dd7dd86f80c76b39329f3b"
}
