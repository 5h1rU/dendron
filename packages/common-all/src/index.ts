export * from "./types";
export * from "./store";
export * from "./node";
export * from "./helpers";
