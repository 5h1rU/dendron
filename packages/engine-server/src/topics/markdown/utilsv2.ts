import {
  DLink,
  DNoteLoc,
  NotePropsV2,
  NoteUtilsV2,
} from "@dendronhq/common-all";
import _ from "lodash";
import remark from "remark";
import frontmatterPlugin from "remark-frontmatter";
import markdownParse from "remark-parse";
import {
  DendronLinksOpts,
  dendronLinksPlugin,
  WikiLinkNote,
} from "./plugins/dendronLinksPlugin";
const selectAll = require("unist-util-select").selectAll;

export class ParserUtilsV2 {
  static createWikiLinkRE(opts?: { oldLink: string }) {
    const { oldLink } = opts || {};
    if (oldLink) {
      const match = ParserUtilsV2.escapeForRegExp(oldLink);
      return `\\[\\[\\s*?(.*\\|)?\\s*${match}\\s*\\]\\]`;
    }
    return "\\[\\[\\s*?(.*\\|)?\\s*(?<name>.*)\\s*\\]\\]";
  }

  static getRemark(opts?: { dendronLinksOpts: DendronLinksOpts }) {
    const { dendronLinksOpts } = _.defaults(opts, { dendronLinksOpts: {} });
    return remark()
      .use(markdownParse, { gfm: true })
      .use(frontmatterPlugin, ["yaml"])
      .use(dendronLinksPlugin, dendronLinksOpts)
      .use({ settings: { listItemIndent: "1", fences: true } });
  }

  // static findLinks({ note }: { note: NotePropsV2 }): DLink[] {
  //   const content = note.body;
  //   const reWiki = ParserUtilsV2.createWikiLinkRE();
  //   const wikiLinks = content.matchAll(new RegExp(reWiki, "gi"));
  //   const wikiLinksMatches = Array.from(wikiLinks, (m) => {
  //     if (_.isUndefined(m.index)) {
  //       throw Error("no index found, findLinks");
  //     }
  //     // will barf at `[[]]` case
  //     if (!m.groups?.name) {
  //       throw Error(`no name found, findLInks, ${JSON.stringify(note)}`);
  //     }
  //     return {
  //       match: m[0],
  //       value: m.groups.name,
  //       alias: m.groups?.alias,
  //       start: m.index,
  //       end: m.index + m[0].length,
  //     };
  //   });
  //   return wikiLinksMatches.map((m) => ({
  //     type: "wiki",
  //     from: NoteUtilsV2.toLoc(note),
  //     original: m.match,
  //     value: m.value,
  //     alias: m?.alias,
  //     pos: { start: m.start, end: m.end },
  //     to: {
  //       fname: m.value,
  //     },
  //   }));
  // }

  static findLinks({ note }: { note: NotePropsV2 }): DLink[] {
    const content = note.body;
    let remark = ParserUtilsV2.getRemark();
    //const out = remark.processSync(content);
    let out = remark.parse(content);
    let out2: WikiLinkNote[] = selectAll("wikiLink", out);
    const dlinks = out2.map(
      (m: WikiLinkNote) =>
        ({
          type: "wiki",
          from: NoteUtilsV2.toLoc(note),
          original: m.value,
          value: m.value,
          alias: m.data.alias,
          pos: { start: m.position?.start.offset, end: m.position?.end.offset },
          to: {
            fname: m.value,
          },
        } as DLink)
    );
    return dlinks as DLink[];
  }

  static escapeForRegExp(value: string) {
    return value.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
  }

  static async replaceLinks(opts: {
    content: string;
    from: DNoteLoc;
    to: DNoteLoc;
  }) {
    const { content, from, to } = opts;
    let remark = ParserUtilsV2.getRemark({
      dendronLinksOpts: {
        replaceLink: { from, to },
      },
    });
    const out = await remark.process(content);
    return out.toString();
  }
}
