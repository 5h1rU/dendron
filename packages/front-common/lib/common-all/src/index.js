export * from "./node";
export * from "./helpers";
export * from "./integrations/dropbox";
//# sourceMappingURL=index.js.map