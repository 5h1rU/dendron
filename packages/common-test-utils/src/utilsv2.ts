import { NoteUtilsV2, SchemaUtilsV2 } from "@dendronhq/common-all";
import {
  note2File,
  schemaModuleOpts2File,
  tmpDir,
} from "@dendronhq/common-server";
import _ from "lodash";
import {
  CreateEngineFunction,
  EngineTestUtilsV4,
  RunEngineTestFunction,
  RunEngineTestFunctionV4,
  runJestHarnessV2,
  SetupVaultsOptsV4,
} from ".";
import {
  PostSetupHookFunction,
  PreSetupHookFunction,
  SetupHookFunction,
} from "./types";

export class TestPresetEntryV4 {
  public preSetupHook: PreSetupHookFunction;
  public testFunc: RunEngineTestFunctionV4;

  constructor(
    func: RunEngineTestFunctionV4,
    opts?: {
      preSetupHook?: PreSetupHookFunction;
    }
  ) {
    let { preSetupHook } = opts || {};
    this.preSetupHook = preSetupHook ? preSetupHook : async () => {};
    this.testFunc = func;
  }
}

/**
 * Run engine test with relative vaults
 */
export async function runEngineTestV4(
  func: RunEngineTestFunction,
  opts: {
    preSetupHook?: SetupHookFunction;
    postSetupHook?: PostSetupHookFunction;
    createEngine: CreateEngineFunction;
    expect: any;
  }
) {
  const { preSetupHook, createEngine } = _.defaults(opts, {
    preSetupHook: async ({}) => {},
    postSetupHook: async ({}) => {},
  });

  // setup root and vaults
  const wsRoot = tmpDir().name;
  const setupVaultsOpts: SetupVaultsOptsV4[] = ["vault1", "vault2"].map(
    (ent) => ({
      vault: { fsPath: ent },
      preSetupHook: async ({ vpath, vault, wsRoot }) => {
        const rootModule = SchemaUtilsV2.createRootModule({
          created: "1",
          updated: "1",
          vault,
        });
        await schemaModuleOpts2File(rootModule, vpath, "root");

        const rootNote = await NoteUtilsV2.createRoot({
          created: "1",
          updated: "1",
          vault,
        });
        await note2File({ note: rootNote, vault, wsRoot });
      },
    })
  );
  const resp = await EngineTestUtilsV4.setupWS({ wsRoot, setupVaultsOpts });

  await preSetupHook({ wsRoot, vaults: resp.vaults });
  const engine = createEngine({ wsRoot, vaults: resp.vaults });
  await engine.init();
  // const resp = await postSetupHook({wsRoot, vaults, engine})
  const results = await func({ wsRoot, vaults: resp.vaults, engine });

  await runJestHarnessV2(results, expect);
}
