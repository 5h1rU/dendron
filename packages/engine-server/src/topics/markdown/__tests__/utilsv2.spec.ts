import { DNoteLoc, NoteUtilsV2 } from "@dendronhq/common-all";
import _ from "lodash";
import { ParserUtilsV2 } from "../utilsv2";

describe(ParserUtilsV2, () => {
  describe("findLinks", async () => {
    test("one link", () => {
      const note = NoteUtilsV2.create({
        fname: "foo",
        id: "foo",
        created: "1",
        updated: "1",
        body: "[[bar]]",
      });
      const links = ParserUtilsV2.findLinks({ note });
      expect(links).toMatchSnapshot("bond");
      expect(links[0].to?.fname).toEqual("bar");
    });
  });

  describe("replaceLinks", async () => {
    test("one", async () => {
      const content = ["[[bond]]"].join("\n");
      const from: DNoteLoc = {
        fname: "bond",
        vault: {
          fsPath: "/tmp",
        },
      };
      const to: DNoteLoc = {
        fname: "bar",
        vault: {
          fsPath: "/tmp",
        },
      };
      const out = await ParserUtilsV2.replaceLinks({ content, from, to });
      expect(out).toMatchSnapshot();
      expect(_.trim(out)).toEqual("[[bar]]");
    });

    test("multiple", async () => {
      const content = ["[[bond]]", "[[bond]]"].join("\n");
      const from: DNoteLoc = {
        fname: "bond",
        vault: {
          fsPath: "/tmp",
        },
      };
      const to: DNoteLoc = {
        fname: "bar",
        vault: {
          fsPath: "/tmp",
        },
      };
      const out = await ParserUtilsV2.replaceLinks({ content, from, to });
      expect(out).toMatchSnapshot();
    });

    test("inline code", async () => {
      const content = ["`[[bond]]`"].join("\n");
      const from: DNoteLoc = {
        fname: "bond",
        vault: {
          fsPath: "/tmp",
        },
      };
      const to: DNoteLoc = {
        fname: "bar",
        vault: {
          fsPath: "/tmp",
        },
      };
      const out = await ParserUtilsV2.replaceLinks({ content, from, to });
      expect(out).toMatchSnapshot();
      expect(_.trim(out)).toEqual("`[[bond]]`");
    });

    test("fenced code", async () => {
      const content = ["```", "[[bond]]", "```"].join("\n");
      const from: DNoteLoc = {
        fname: "bond",
        vault: {
          fsPath: "/tmp",
        },
      };
      const to: DNoteLoc = {
        fname: "bar",
        vault: {
          fsPath: "/tmp",
        },
      };
      const out = await ParserUtilsV2.replaceLinks({ content, from, to });
      expect(out).toMatchSnapshot();
    });

    test("with alias", async () => {
      const content = ["[[hero|bond]]"].join("\n");
      const from: DNoteLoc = {
        fname: "bond",
        vault: {
          fsPath: "/tmp",
        },
      };
      const to: DNoteLoc = {
        fname: "bar",
        vault: {
          fsPath: "/tmp",
        },
      };
      const out = await ParserUtilsV2.replaceLinks({ content, from, to });
      expect(out).toMatchSnapshot();
      expect(_.trim(out)).toEqual("[[hero|bar]]");
    });

    test("with offset ", async () => {
      const content = ["   [[bond]]"].join("\n");
      const from: DNoteLoc = {
        fname: "bond",
        vault: {
          fsPath: "/tmp",
        },
      };
      const to: DNoteLoc = {
        fname: "bar",
        vault: {
          fsPath: "/tmp",
        },
      };
      const out = await ParserUtilsV2.replaceLinks({ content, from, to });
      expect(out).toMatchSnapshot();
      expect(_.trim(out)).toEqual("[[bar]]");
    });
  });
});
