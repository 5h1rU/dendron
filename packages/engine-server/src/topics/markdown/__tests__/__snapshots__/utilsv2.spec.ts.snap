// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParserUtilsV2 findLinks one link: bond 1`] = `
Array [
  Object {
    "alias": undefined,
    "from": Object {
      "fname": "foo",
      "id": "foo",
    },
    "original": "[[bar]]",
    "pos": Object {
      "end": 7,
      "start": 0,
    },
    "to": Object {
      "fname": "bar",
    },
    "type": "wiki",
    "value": "bar",
  },
]
`;

exports[`ParserUtilsV2 replaceLinks fenced code 1`] = `
"\`\`\`
[[bond]]
\`\`\`
"
`;

exports[`ParserUtilsV2 replaceLinks inline code 1`] = `
"\`[[bond]]\`
"
`;

exports[`ParserUtilsV2 replaceLinks multiple 1`] = `
"[[bar]]
[[bar]]
"
`;

exports[`ParserUtilsV2 replaceLinks one 1`] = `
"[[bar]]
"
`;

exports[`ParserUtilsV2 replaceLinks with alias 1`] = `
"[[hero|bar]]
"
`;

exports[`ParserUtilsV2 replaceLinks with offset  1`] = `
"   [[bar]]
"
`;
